setup.sh

```diff
@@ -92,8 +93,20 @@ function check_dependencies()
     else
       eval "sudo $cmd"
     fi
-  fi
 
+    # installation failed...
+    local ret="$?"
+    if [[ "$ret" -ne 0 ]]; then
+      local errmsg=''
+      errmsg="[ERROR]:Dependency installation failure: "
+      [[ "$ret" -eq 1 ]] && errmsg+="lacked sudo privileges to install the packages "
+      [[ "$ret" -ne 1 ]] && errmsg+="something went wrong when installing "
+      errmsg+=$package_list
```

Comment:
Use double-quotes around `$package_list`. 

**Suggestion**: add a line break before `$package_list`, like

```bash
errmsg+=$'\n'"$package_list"
```

to make a more organized output to the user.